### PR TITLE
add debug flag, debug logging

### PR DESCRIPTION
**Pull request by dt-rush**
Connects-to: #181
Connects-to: #221
Connects-to: #222
Change-type: minor
Signed-off-by: dt-rush <nickp@balena.io>

### CODE REVIEW

File: src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 import org.yaml.snakeyaml.Yaml;
 
 public class CloudWatchCollector extends Collector {
-    private static final Logger LOGGER = Logger.getLogger(CloudWatchCollector.class.getName());
+    private static final Logger LOGGER = Logger.getLogger(CloudWatchCollector.class.getPackage().getName());
```
Comment (brian-brazil) [inline src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java]: Why the getPackage?

File: src/main/java/io/prometheus/cloudwatch/WebServer.java [hunk 2]
```diff
@@ -9,40 +14,80 @@
...
+      }
+      LOGGER.log(Level.FINEST, "debug logging is on");
+    }
+
+    private static void printUsage() {
+        System.out.println("cloudwatch_exporter [-d] <port> <yml configuration file>");
+        System.out.println("-d,--debug     enable debug logs");
+    }
+
+    private static void parseOptions(String[] args) {
```
Comment (brian-brazil) [inline src/main/java/io/prometheus/cloudwatch/WebServer.java]: No need for all this, the user can create a logging.properties.

Comment (alexives) [reply src/main/java/io/prometheus/cloudwatch/WebServer.java]: It's actually kind of nice to be able to pass this in as an option when you're running in a container. Then you don't have to worry about mounting another config file somewhere.

**Comment by carlosjgp** [comment]
Any news about this?

**Comment by willejs** [comment]
@brian-brazil @dt-rush any chance you could look at this again? It would be great to see it get merged in, its hard to debug right now.

File: src/main/java/io/prometheus/cloudwatch/WebServer.java [hunk 3]
```diff
@@ -9,40 +14,80 @@
...
+          context.setContextPath("/");
+          server.setHandler(context);
+          context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
+          context.addServlet(new ServletHolder(new DynamicReloadServlet(collector)), "/-/reload");
+          context.addServlet(new ServletHolder(new HealthServlet()), "/-/healthy");
+          context.addServlet(new ServletHolder(new HealthServlet()), "/-/ready");
+          context.addServlet(new ServletHolder(new HomePageServlet()), "/");
+          server.start();
+        } catch (Exception e) {
+          LOGGER.log(Level.SEVERE, String.format("failed to start sever on port %d", port));
```
Comment (rothsa) [inline src/main/java/io/prometheus/cloudwatch/WebServer.java:84]: server not sever

Comment (alkanen) [reply src/main/java/io/prometheus/cloudwatch/WebServer.java:84]: > server not sever

No, the log level is called "SEVERE"

Comment (rothsa) [reply src/main/java/io/prometheus/cloudwatch/WebServer.java:84]: I'm not referring to SEVERE, but sever: "failed to start sever on port %d"

**Comment by rothsa** [comment]
@dt-rush I'm working on a couple additional logging points, and have added in changes to address the above comments and master which has advanced since this pull was made. Shall I send the PR to the balena fork? I will create a new pull in this main repo including your commits, if I do not hear from you.

File: src/main/java/io/prometheus/cloudwatch/WebServer.java (hunk 3 above)
Comment (alkanen) [reply src/main/java/io/prometheus/cloudwatch/WebServer.java:84]: Haha, sorry, my bad.

Comment (alkanen) [inline src/main/java/io/prometheus/cloudwatch/WebServer.java:84]:
```suggestion
          LOGGER.log(Level.SEVERE, String.format("failed to start server on port %d", port));
```

Comment (alkanen) [reply src/main/java/io/prometheus/cloudwatch/WebServer.java:84]: I made a change suggestion on the typo.

**Comment by rothsa** [comment]
Link to the related discussion on prometheus developers mailing list: https://groups.google.com/d/msgid/prometheus-developers/115d3e68-4696-457c-8e41-8d2dcd6ab1b5%40googlegroups.com.

File: pom.xml [hunk 4]
```diff
@@ -119,6 +119,7 @@
           <archive>
             <manifest>
               <mainClass>io.prometheus.cloudwatch.WebServer</mainClass>
+              <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
```
Comment (brian-brazil) [inline pom.xml:122]: Why is this changing?

File: src/main/java/io/prometheus/cloudwatch/WebServer.java (hunk 2 above)
Comment (brian-brazil) [reply src/main/java/io/prometheus/cloudwatch/WebServer.java:37]: My comment stands, there's no need to slowly reinvent how logging configuration is done in Java. Please remove this.

**Review by brian-brazil — COMMENTED** [review]
If you still wish to add additional logging points, there are still unresolved comments.

**Comment by rothsa** [comment]
No, that's alright, I made the changes for our private fork. OK to close this from my perspective.

**Comment by brian-brazil** [comment]
Okay, some of those log lines would have been useful but I'll close this off then.

**Pull request closed without merging**